addons/event_sale/models/event_registration.py

```diff
@@ -28,8 +28,8 @@ class EventRegistration(models.Model):
     def _compute_registration_status(self):
         self.filtered(lambda reg: not reg.state).state = 'draft'
         for so_line, registrations in self.grouped('sale_order_line_id').items():
-            cancelled_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
-            cancelled_registrations.state = 'cancel'
+            cancelled_registrations = registrations.filtered(lambda reg: reg.state == 'cancel' or reg.sale_order_id.state == 'cancel')
+            cancelled_registrations.filtered(lambda reg: reg.state != 'cancel').write({"state" :'cancel'})
```

Comment:
why not keep `.state = 'cancel'`? in old odoo version this would only work on recordset of one record, but now it should work and have the same effect as `write({"state": "cancel"})`